crates/matrix-sdk/src/event_cache/linked_chunk.rs

```diff
@@ -1347,28 +1270,28 @@ mod tests {
 
     #[test]
     fn test_push_items() {
-        use ListenerUpdate::*;
+        use super::LinkedChunkUpdate::*;
 
-        let mut linked_chunk = LinkedChunk::<3, char, (), _>::new_with_listener(Listener::new());
+        let mut linked_chunk = LinkedChunk::<3, char, ()>::new();
         linked_chunk.push_items_back(['a']);
 
         assert_items_eq!(linked_chunk, ['a']);
         assert_eq!(
-            linked_chunk.listener().take(),
+            linked_chunk.updates().drain(..).collect::<Vec<_>>(),
```

Comment:
I think `.drain(..).collect::<Vec<_>>()` is equivalent to `.take()` here? (Unless you really want to optimize and not clear the capacity of the underlying vector, which I suppose you do not care about in testing :D)

Reply:
I don't want it to be `None`, otherwise it will disable the update history. We want to get a `Some(empty_vec)`.